docs/guides/tutorials-appium/appium-cross-platform-automation.md

```diff
@@ -0,0 +1,159 @@
+
+A good example for using PageObject design pattern will be transformaing this test method from the normal way:
+
+{% highlight java %}
+@Test
+public void twoPlusTwoOperation() {
+    // Get elements from your app screen.
+    MobileElement buttonTwo = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/digit2")));
+    MobileElement buttonPlus = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/plus")));
+    MobileElement buttonEquals = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/equal")));
+    MobileElement resultField = (MobileElement)(driver.findElement(By.xpath("//android.widget.EditText[1]")));
+
+    // Add two and two ( 2 + 2 ).
+    buttonTwo.click();
+    buttonPlus.click();
+    buttonTwo.click();
+    buttonEquals.click();
+
+    // Check if within given time the correct result appears in the designated field ( check if 2 + 2 = 4 ).
+    new WebDriverWait(driver, 30).until(ExpectedConditions.textToBePresentInElement(resultField, 4));
+}
+{% endhighlight %}
+
+To the PageObject design pattern way:
+
+{% highlight java %}
+@Test
+public void twoPlusTwoOperation() {
+    app.calculatorScreen().addTwoAndTwo();
+    assertTrue(app.calculatorScreen().isResultCorrect("4"));
+}
+{% endhighlight %}
+
+For more information about PageObject, see our [Appium setup with PageObject](/docs/guides/appium-advanced-setup/) guide.
+
+<h3 id="appium_anno">Appium annotations</h3>
+It is hard to use the old way of finding elements for a cross-platform project, so we will use <strong>@AndroidFindBy()</strong> and <strong>@iOSFindBy()</strong> Appium annotations to fetch UI elements in our app, using (id, xpath or text, ...). Here in the folowing example we are tring to find an element using its Id on Android:
+
+{% highlight java %}
+@AndroidFindBy(id = "signup_button")
+MobileElement signUpButton;
+{% endhighlight %}
+
+And here we are using the xpath to find an element in an iOS applicatoion:
+
+{% highlight java %}
+@iOSFindBy(xpath = "//UIAButton[1]")
+MobileElement signUpButton;
+{% endhighlight %}
+
+In the case of cross-platform project, we use both of <strong>@AndroidFindBy()</strong> and <strong>@iOSFindBy()</strong> to find element(s):
+
+{% highlight java %}
+@AndroidFindBy(id = "signup_button")
+@iOSFindBy(xpath = "//UIAButton[1]")
+MobileElement signUpButton;
+{% endhighlight %}
+
+Which annotation is going to be used? When we use Android and iOS annotations together, it doesn't at all mean that both of them are going to be used at the same time, only one of them will be, and it depends on the platform we use. So if we are testing on Android platform and instantiating an Android Appium Driver in our setup method, then the Android annotation will be used, and the iOS annotation will be also used if we instantiated an iOS Appium Driver.
```

Comment:
"so if we are testing on THE Android platform";
"So if we are testing on Android platform and instantiating an Android Appium Driver in our setup method, then the Android annotation will be used; if we have instantiated an iOS driver, the iOS annotation will be used.